controllers/machinehealthcheck_controller.go

```diff
@@ -24,39 +34,122 @@ type MachineHealthCheckReconciler struct {
 	Recorder                    record.EventRecorder
 	ClusterUpgradeStatusChecker cluster.UpgradeChecker
 	MHCChecker                  mhc.Checker
+	ReconcileMHC                bool
```

Comment:
IIUC this param meant to function as a feature gate, right ?

Reply:
yes, but I'm going to replace with the official feature gate for disabling the MAO MHC controller, like it's done in MAO as well (see https://github.com/openshift/machine-api-operator/pull/1151)